GeneratorInterface/Core/interface/HadronizerFilter.h

```diff
@@ -269,6 +282,25 @@ namespace edm
       finalEvent->weights()[0] *= multihadweight;
     }
     
+    if (secondaryUnweighting_) {
+      double weight = finalGenEventInfo->weights()[0];
+      double absweight = std::abs(weight);
+      if (absweight<maxAbsWeight_) {
+        double randomflat = randomEngineSentry.randomEngine()->flat();
```

Comment:
@Dr15Jones can you comment whether this is an acceptable use of the randomEngineSentry object which is otherwise being used by the hadronizer itself underneath.  (I would like to throw additional random numbers here, and I would like them to be tied to the random seed of the hadronizer)